audio-worklet/bitcrusher-test.html

```diff
@@ -15,6 +15,172 @@
+    	  let delay = context.createDelay();
+    	  delay.delayTime.value = bufferLength / context.sampleRate;
+
+    	  oscillator.connect(bitcrusher.input);
+    	  bitcrusher.output.connect(merger, 0, 0);
+    	  oscillator.connect(delay).connect(merger, 0, 1);
+    	  merger.connect(context.destination);
+
+    	  // When audio buffer is ready, verify bitcrushed samples are unaltered.
+    	  context.startRendering()
+    	      .then((buffer) => {
+    	        let bitcrusherOutput = buffer.getChannelData(0);
+    	        let originalOutput = buffer.getChannelData(1);
+
+    	        // Allow for fractional error beyond audible perception. This 
+    	        // error occurs because any sample passing through bitcrusher will 
+    	        // undergo manipulation in Math.round() and will therefore be 
+    	        // represented by a new floating point number that differs     	       // slightly from the original.
+    	        // In tested samples and at a sample rate of 44,1k, the 
+    	        // maximum observed error is 6.183981895446777e-7.
+    	        
+    	        const permittedSampleError = 6.1840e-7;
+    	        let maxError = 0;
+    	        for (let i = 0; i < bitcrusherOutput.length; i++) {
+    	          let diff = Math.abs(bitcrusherOutput[i] - originalOutput);
+    	          if (diff > maxError) maxError = diff;  
+    	        }
+    	        console.assert(
+    	            maxError < permittedSampleError,
+    	            'The maximum error between bitcrusher and original output, ' +
+    	                maxError + ', exceeds the maximum permitted error, ' +
+    	                permittedSampleError);
+
+    	        // Verify samples from unadultered oscillator match samples from
+    	        // bitcrushed oscillator with non-information reducing parameters.
+    	        for (let i = 0; i < originalOutput.length; i++) {
+    	          let crushedSample = bitcrusherOutput[i];
+    	          let originalSample = originalOutput[i];
+    	          const diff = Math.abs(originalSample - crushedSample);
+    	          
+    	          if (i < bufferLength) {
+    	            console.assert(
+    	                bitcrusherOutput[i] == 0,
+    	                'Bitcrusher sample at ' + i + 'expected to be 0 but it'
+    	                + ' was ' + bitcrusherOutput[i]);
+    	          }
+    	          console.assert(
+    	              diff < permittedSampleError,
+    	              'Bitcrushed sample at ' + i + ' is ' + crushedSample + 
+    	              ' but ' + originalSample + ' for oscillator');
+    	        }
+    	      });
+    	}
+
+    	/**
+    	 * Verify that bitcrusher handles illegal input correctly.
+    	 */
+    	function testIllegalInput() {
+    	  try {
+    	    let bitcrusher = new Bitcrusher(0);
+    	    console.error('Bitcrusher did not throw illegal context error.');
+    	  }
+    	  catch (error) {}
```

Comment:
I haven't been printing any messages when tests pass. Should I change this in other places too?